gen/calc.py

```diff
@@ -672,13 +672,15 @@ def calculate_check_config(check_time):
             'checks': {
                 'components_master': {
                     'description': 'All DC/OS components are healthy.',
-                    'cmd': ['/opt/mesosphere/bin/dcos-checks', '--role', 'master', 'components'],
+                    'cmd': ['/opt/mesosphere/bin/dcos-checks', '--role', 'master', 'components',
+                            '--exclude', 'dcos-checks-poststart.service', 'dcos-checks-poststart.timer'],
                     'timeout': '3s',
                     'roles': ['master']
                 },
                 'components_agent': {
                     'description': 'All DC/OS components are healthy',
-                    'cmd': ['/opt/mesosphere/bin/dcos-checks', '--role', 'agent', 'components', '--port', '61001'],
+                    'cmd': ['/opt/mesosphere/bin/dcos-checks', '--role', 'agent', 'components', '--port', '61001',
+                            '--exclude', 'dcos-checks-poststart.service', 'dcos-checks-poststart.timer'],
```

Comment:
This `--exclude` flag only respects the unit that immediately follows it. If the second unit listed here (the timer) is failing, then this check will fail, when it should be ignoring it. I'm not sure how to provoke a timer unit to fail, but I just swapped the order of the unit names in check config, failed the service unit, and then this check failed to ignore it. 

So this is a bug merged in with https://github.com/dcos/dcos-checks/pull/22. Is there something we could have done to better test the new flag in that PR?

Reply:
The unit test has this case covered - https://github.com/dcos/dcos-checks/blob/master/cmd/checks/components/components_check_test.go#L79 . The response has both failing, it could be the way its getting passed here. good catch. ill look into this.

Reply:
The unit test didn't catch this error, which means it doesn't have this case covered. The test is calling the backing function directly, rather than invoking the CLI (and the code that parses CLI params).